pintc/src/pint_parser.lalrpop

```diff
@@ -227,6 +227,15 @@ LetDecl: () = {
             .insert_variable(handler, mod_prefix, name.1, &name.0, ty, init, span)
             .map(|_| ());
     },
+    <l:@L> "let" <name:LetName> <ty:(":" <Type>)?> "=" <init:AsmBlockExpr> <r:@R> => {
```

Comment:
For `asm` blocks the type shouldn't be optional.

Reply:
I wonder without actually trying if we have mandatory types for ASM `let`s then we could perhaps type them with a quick pass in the type checker and then avoid the trickiness added in this PR for inference and the ASM comment in #1079.

Reply:
Done.